File: 0200-number-of-islands/0200-number-of-islands.cpp
class Solution {
public:

    int total = 0;

    int numIslands(vector<vector<char>>& grid) {
        int bL = 0;
        int bU = 0;
        int bR = grid[0].size()-1;
        int bD = grid.size()-1;
        for(int i = 0;i<=bD;i++){
            for(int j = 0;j<=bR;j++){
                if(grid[i][j] == '1'){
                    total+=1;
                    dive(grid,i,j,bL,bU,bR,bD);
                }
            }
        }
        return total;
    }

    void dive(vector<vector<char>>& grid,int i, int j, int bL, int bU, int bR, int bD){
        if(i<bU || i>bD || j>bR || j<bL){
            return;
        }
        if(grid[i][j] == '1'){
            grid[i][j] = '0';
        dive(grid,i+1,j,bL,bU,bR,bD);
        dive(grid,i-1,j,bL,bU,bR,bD);
        dive(grid,i,j+1,bL,bU,bR,bD);
        dive(grid,i,j-1,bL,bU,bR,bD);
        }
        
    }
};

File: 0200-number-of-islands/0200-number-of-islands.java
class Solution {
    public void islands(char[][] grid, int bD, int bR, int i, int j){
        if(i>=0 && j>=0 && i<=bD && j<=bR){
            if(grid[i][j]=='1'){
                    grid[i][j]='0';
                
                    islands(grid,bD,bR,i+1,j);
                    islands(grid,bD,bR,i-1,j);
                    islands(grid,bD,bR,i,j+1);
                    islands(grid,bD,bR,i,j-1);
                
            }
        }
    }


    public int numIslands(char[][] grid) {
        int bD = grid.length-1;
        int bR = grid[0].length-1;
        int prevSetSize = 0;
        int ans = 0;
        for(int i = 0;i<=bD;i++){
            for(int j = 0;j<=bR;j++){
                if(grid[i][j]=='1'){
                    ans++;
                    islands(grid,bD,bR,i,j);
                    
                }
            }
        }
        return ans;
    }
}